### DISCUSSION TITLE
How to change TileJSON URL base server

### DISCUSSION THREAD

**Discussion by zmatez** [maplibre/martin#1558] [General]
Hi, in my app I do proxy (using Hono) to verify if user has access to requested data (using a middleware and token cookie):
```ts
router.get('/main/:z?/:x?/:y?',async (c) => {
    const {z,x,y} = c.req.param();
    let path = '';
    if(z != null && x != null && y != null) {
        path = '/' + z + '/' + x + '/' + y;
    }
    return fetch('http://localhost:8000/europe' + path)
})
```
(on localhost:8000 is my Martin server)

However what Martin returns in tilejson looks like:
```"tilejson":"3.0.0","tiles":["http://localhost:8000/europe/{z}/{x}/{y}"]```
So it's still localhost:8000 and my clients try to connect to that url, but should to that outgoing hono's app (so like `https://maps.example.com/map/main/z/x/y`)
How to tell Martin that the data is available from another URL for the world and it should send the correct address?
Thanks

**Comment by CommanderStorm** [comment]
To change the url, you need to change the request headers as for example with nginx:
https://maplibre.org/martin/run-with-nginx.html#rewriting-urls

There is also the `base_url` configuration option, but that only changes the path, not the host.

**Reply by zmatez** [reply]
Yea it worked, thanks!

if someone needs:
```ts
router.get('/main/:z?/:x?/:y?',async (c) => {
    const {z,x,y} = c.req.param();
    let path = '';
    if(z != null && x != null && y != null) {
        path = '/' + z + '/' + x + '/' + y;
    }

    const headers = {
        'X-Rewrite-URL': c.req.url,
        'X-Forwarded-Host': `${c.req.header('host')}`,
        'X-Forwarded-Proto': 'https'
    };

    return fetch('http://localhost:8000/europe' + path, {
        method: 'get',
        headers: headers
    })
})```
